core/src/main/java/com/linecorp/armeria/client/retry/RetryRuleUtil.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.retry;
+
+final class RetryRuleUtil {
+
+    static RetryStrategy toRetryStrategy(RetryRule rule) {
+        return (ctx, cause) -> rule.shouldRetry(ctx, cause).thenApply(RetryRuleDecision::backoff);
+    }
+
+    static RetryRule fromRetryStrategy(RetryStrategy strategy) {
+        return (ctx, req) -> strategy.shouldRetry(ctx, req).thenApply(RetryRuleDecision::retry);
```

Comment:
`strategy.shouldRetry(ctx, req)` could return `null`, so I think we should handle that.

Reply:
RetryClient does not check null of the return value. So I thought `RetryStrategy.shouldRetry(...) must not return null value.
https://github.com/line/armeria/blob/fbe305f15310445337ce42e7bf6ad40ffb3219c1/core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java#L267
Do I need to update RetryClient too? 🤔

Reply:
What I meant was when the `backoff` is `null`: https://github.com/line/armeria/blob/a9544c21e9a60b9bc689eab4a6c7094b4e67e16f/core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java#L255
If it's `null`, then NPE is raised here: https://github.com/line/armeria/pull/2661/files#diff-b84a9cb0a39bd7bd59e96e1a4e2b5c83R39
Right?

Reply:
I see. 😅